video/out/vo_gpu_next.c

```diff
@@ -368,7 +371,8 @@ static void update_overlays(struct vo *vo, struct mp_osd_res res,
                 // Seems like HDR subtitles are targeting SDR white
                 if (pl_color_transfer_is_hdr(ol->color.transfer)) {
                     ol->color.hdr = (struct pl_hdr_metadata) {
-                        .max_luma = PL_COLOR_SDR_WHITE,
+                        .max_luma = (!target_hint || p->next_opts->image_subs_hdr_peak)
+                                    ? PL_COLOR_SDR_WHITE : 0.0f,
```

Comment:
What it does in practice is to set subtitles to 10000 nits, which is almost certainly invalid. I'm not sure why you want to have broken subtitle rendering.

Either way, we can have an option `--image-subs-hdr-peak=<video,sdr>` which would actually not set `PL_COLOR_SDR_WHITE` for image subs and reuse video. I believe some may be mastered like that, but in my testing they were mostly SDR white.

Also please remove all `target_hint` checks, it's not correct way to check if we are outputting HDR. Users can adjust this new option in config profile, if they like.

Reply:
You basically do the same few lines below
```
diff --git a/video/out/vo_gpu_next.c b/video/out/vo_gpu_next.c
index f94671e89b..0e7ada500a 100644
--- a/video/out/vo_gpu_next.c
+++ b/video/out/vo_gpu_next.c
@@ -386,6 +386,9 @@ static void update_overlays(struct vo *vo, struct mp_osd_res res,
         case SUBBITMAP_LIBASS:
             if (src && item->video_color_space && !pl_color_space_is_hdr(&src->params.color))
                 ol->color = src->params.color;
+            ol->color.hdr = (struct pl_hdr_metadata) {
+                .max_luma = 1111,
+            };
             ol->mode = PL_OVERLAY_MONOCHROME;
             ol->repr.alpha = PL_ALPHA_INDEPENDENT;
             break;
```

Reply:
One more question: the `sdr` in <video,sdr> corresponds to `PL_COLOR_SDR_WHIT`, what value should `video` correspond to? A dynamic value?

Reply:
> what value should video correspond to? A dynamic value?

Not overridden, it's already copied from source `ol->color = src->params.color`.

Reply:
Why not adjust subtitles when tone mapping? Isn't it the same to adjust subs?

Reply:
> Why not adjust subtitles when tone mapping? Isn't it the same to adjust subs?
> 
> EDIT: HDR output should work the same as SDR tonemapped output. Hence why the subtitles should be adjusted too, I think. You could check if source is HDR and enable it in this case if you don't want to touch SDR.

Because it would bring back the issue: https://github.com/mpv-player/mpv/issues/8325.
The original purpose of adding these options was for users to adjust the subtitle brightness in HDR output to avoid potentially overbright subtitle displays, it should not affect the tone mapping in SDR output as SDR is unlikely to display overbright subtitles.

I want to insist on this point: the new behavior should not bring back possible problems.

Reply:
Ok, I don't mind either way, as long as current defaults doesn't change.